jhputnam/mailroom.py

```diff
@@ -2,127 +2,155 @@
-          "Kayaba Akihiko": [10000.00, 3853.00, 6147.00],
-          "Doctor Evil": [1000000.00]}
 
-
-def menu():
+def initdb():
     """
-    Function to show the main menu.
-    :return: none
+    This function initializes the donor db using pickle.
+    returns donor dictionary.
     """
 
-    # Originally put the menu in main() but ran into a roadblock when
-    # the lab asked to return to the menu. Not sure how to do that in main
-    # without getting into trouble.
+    try:
+        with open('donordb.dat', 'rb') as handle:
+            donors = pickle.load(handle)
 
-    print("Welcome to mailroom. Please pick one of the following options:\n")
-    print("1. Send a thank you")
-    print("2. Generate report")
-    print("3. Quit\n")
+    except FileNotFoundError:
+        print("Database file 'donordb.dat' not found!"
+              " What would you like to do?\n")
+        print("1. Create new database.")
+        print("2. Abort.\n")
+        print()
 
-    while True:
-        try:
-            choice = int(input("Enter 1, 2, or 3 from the menu above: "))
-            if choice not in (1, 2, 3):
+        while True:
+            choice = input("Enter option 1 or 2: ")
+            if choice not in ('1', '2'):
                 print("Not a valid option. Try again.")
                 continue
             break
-        except ValueError:
-            print("Not a valid option. Try again.")
 
-    if choice == 1:
-        send_thank_you()
-    elif choice == 2:
-        generate_report()
-    else:
-        print("Goodbye!")
+        if choice == '1':
+            donors = {}
+            updatedb(donors)
+
+        elif choice == '2':
+            print("Goodbye!")
+            sys.exit()
+
+    return donors
 
 
-def send_thank_you():
+def updatedb(donors):
     """
-    Function to handle both adding new users and to initiate the sending
-    of a thank you email to the donor.
-    :return: none
+    Function to write the donor dictionary out to a pickle file.
+    """
+
+    try:
+        with open('donordb.dat', 'wb') as handle:
+            pickle.dump(donors, handle, protocol=pickle.HIGHEST_PROTOCOL)
+    except IOError:
+        raise "Error writing to the donors file. Please check permissions."
+
+
+def add_donation(donors):
+    """
+    Function to handle adding donations.
+    Arguments:
+    donors: donors dictionary object.
     """
 
     while True:
-        fullname = input("Please enter the donor's full name or type 'list' "
-                         "to see a list of donors: ")
+        fullname = input("Please enter the donor's full name or type 'list'"
+                         " to see a list of donors: ")
 
-        # Listing is not currnetly pretty.
-        # Might clean up the presentation if time allows.
         if fullname == "list":
-            for name in DONORS:
+            for name in donors:
                 print(name)
                 continue
         else:
```

Comment:
Sure, but would that not print the actual dictionary data structure? Not exactly what I want.